.github/workflows/main.yml

```diff
@@ -57,9 +57,10 @@ jobs:
     strategy:
       matrix:
         rust: [stable]
-        python-version: [3.8, 3.9, "3.10", "3.11", "3.12"]
+        python-version: [3.8, 3.9, "3.10", "3.11", "3.12.0"]
```

Comment:
I guess this fixes #1041 indirectly

Reply:
It does, although I only did this as an attempt to fix my typo in an earlier iteration and it's not strictly necessary. I can revert this and the stestr usage if you'd prefer to do that separately.

Reply:
Maybe let's keep it separate because there will be other 3.12 releases and there could be a regression. Although it is unlikely

Reply:
Reverted in: https://github.com/Qiskit/rustworkx/pull/1077/commits/ecf713b2246fb0d8fa0415fe6e0594cd1619a6ac